Tools/DataStorage/DataStorage.cs

```diff
@@ -1638,5 +1638,198 @@ public bool RemoveComments(Comment c)
                 myConn?.Close();
             }
         }
+
+        //----------------Movement-----------------
+        public List<Movement> GetMovements(Class c)
+        {
+            string sql1 =
+                "SELECT motion_id, s.student_id, s.surname, s.st_name, s.patronymic, sch_city, sch_region, sch_country " +
+                "FROM(motion INNER JOIN students s on motion.income_st_id = s.student_id) " +
+                " INNER JOIN classes_students cs ON s.student_id = cs.student_id " +
+                $"WHERE cs.class_id='{c.ClassId}'";
+
+            string sql2 =
+                "SELECT motion_id, s.student_id, s.surname, s.st_name, s.patronymic, sch_city, sch_region, sch_country " +
+                "FROM(motion INNER JOIN students s on motion.outsome_st_id = s.student_id) " +
+                " INNER JOIN classes_students cs ON s.student_id = cs.student_id " +
+                $"WHERE cs.class_id='{c.ClassId}'";
+
+            List<Movement> res = new List<Movement>();
+            SqlConnection myConn = new SqlConnection(StationManager.ConnectionString);
+            try
+            {
+                myConn.Open();
+
+                using (SqlCommand command = new SqlCommand(sql1, myConn))
+                {
+                    var reader = command.ExecuteReader();
+                    while (reader.Read())
+                    {
+                        Movement cur = new Movement();
+                        cur.Id = reader.GetInt64(0).ToString();
+                        cur.StudentId = reader.GetInt64(1).ToString();
+                        string f = reader.GetString(2);
```

Comment:
and here to write not cur.StudentId but  cur.Student.StudentId?

Reply:
But we'll also need to fill other props in Student as well. That's the problem.